### PR TITLE
DESTROYS DISARMING

### PR DESCRIPTION
**Pull request by Geevies**
* Disarming now takes a portion of your stamina. You cannot disarm if your stamina is near-empty.
* Disarming someone with a weapon in their hand has a chance for them to retaliate by bashing you with it.

I coded this at 1 AM. I want reviewers to look at this critically. In my testing, the stamina worked out pretty well, you could do around 6 disarms before being pooped. Also in my testing, the retaliation never happened, could be bad luck, could be bad code. Anyone got any ideas? I'd love to improve this.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Pratepresidenten** [comment]
Fuck yeah delete disarming!

**Comment by runecap** [comment]
Does it take the same stamina as what's used to run/sprint?

**Comment by Geevies** [comment]
> 
> 
> Does it take the same stamina as what's used to run/sprint?

`M.stamina = M.stamina - (M.max_stamina / 6)`

It uses a sixth of your stamina per disarm. This of course scales, so, even if you're a species with more stamina, you'll only get about six in before your stamina's drained. 

Unless you're asking whether it uses the same RESOURCE pool, in which case, yes. You won't be able to sprint very well after exerting yourself with four disarm attempts.

**Comment by Geevies** [comment]
And, before the argument pops up, no, I don't think implementing stamina use to normal Harm intent combat is a good idea. It's only being applied to disarm because disarming needs some sort of restraining.

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 1]
```diff
@@ -333,6 +333,10 @@
 			apply_damage(real_damage, hit_dam_type, hit_zone, armour, sharp=is_sharp, edge=is_edge)
 
 		if(I_DISARM)
+			if(M.stamina <= 5)
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: The actual cost subtracted is `M.max_stamina / 6` but the cost to actually _use_ is just. 5? Wat.

Use a var to hold the cost and then reference this.
```dm
var/disarm_cost = M.max_stamina / 6
if (M.stamina <= disarm_cost)
    return
```

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 2]
```diff
@@ -346,22 +350,29 @@
 			msg_admin_attack("[key_name(M)] disarmed [src.name] ([src.ckey]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[M.x];Y=[M.y];Z=[M.z]'>JMP</a>)",ckey=key_name(M),ckey_target=key_name(src))
 			M.do_attack_animation(src)
 
+			M.stamina = M.stamina - (M.max_stamina / 6) //attempting to knock something out of someone's hands, or pushing them over, is exhausting!
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]:
```suggestion
			M.stamina = M.stamina - (M.max_stamina / 6) //attempting to knock something out of someone's hands, or pushing them over, is exhausting!
			M.stamina = Clamp(M.stamina, 0, M.max_stamina)
```
Just to be 100% sure.

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 3]
```diff
@@ -346,22 +352,30 @@
...
 				w_uniform.add_fingerprint(M)
 			var/obj/item/organ/external/affecting = get_organ(ran_zone(M.zone_sel.selecting))
 
-			var/list/holding = list(get_active_hand() = 40, get_inactive_hand = 20)
+			var/list/holding = list(get_active_hand() = 40, get_inactive_hand() = 20)
 
-			//See if they have any guns that might go off
-			for(var/obj/item/weapon/gun/W in holding)
+			//See if they have any weapons to retaliate with
+			for(var/obj/item/weapon/W in holding)
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: Please. PLEASE make this depend on intent. I don't want to decapitate someone for lightly shoving me.

**Comment by Alberyk** [comment]
Needs to fix this #7139
This also affects species like golems and skeletons.

**Comment by skull132** [comment]
Test merged over last weekend, looked good.

**Pull request merged**